lib/iris/tests/unit/analysis/regrid/test__CurvilinearRegridder.py

```diff
@@ -168,6 +169,62 @@ def test_caching(self):
             ],
         )
 
+    def test_derived_coords(self):
+        src = realistic_4d()[0]
+        tgt = realistic_4d()
+        new_lon, new_lat = np.meshgrid(
+            src.coord("grid_longitude").points,
+            src.coord("grid_latitude").points,
+        )
+        coord_system = src.coord("grid_latitude").coord_system
+        lat = AuxCoord(
+            new_lat, standard_name="latitude", coord_system=coord_system
+        )
+        lon = AuxCoord(
+            new_lon, standard_name="longitude", coord_system=coord_system
+        )
+        lat_t = AuxCoord(
+            new_lat.T, standard_name="latitude", coord_system=coord_system
+        )
+        lon_t = AuxCoord(
+            new_lon.T, standard_name="longitude", coord_system=coord_system
+        )
+
+        src.remove_coord("grid_latitude")
+        src.remove_coord("grid_longitude")
+        src_t = src.copy()
+        src.add_aux_coord(lat, [1, 2])
+        src.add_aux_coord(lon, [1, 2])
+        src_t.add_aux_coord(lat_t, [2, 1])
+        src_t.add_aux_coord(lon_t, [2, 1])
+        rg = Regridder(src, tgt)
+        res = rg(src)
+
+        assert len(res.aux_factories) == 1 and isinstance(
+            res.aux_factories[0], HybridHeightFactory
+        )
+        assert np.allclose(
+            res.coord("altitude").points, src.coord("altitude").points
+        )
+        src.transpose([0, 2, 1])
+        res = rg(src)
+        assert np.allclose(
+            res.coord("altitude").points, src.coord("altitude").points
+        )
+
+        rg_t = Regridder(src_t, tgt)
+        res_t = rg_t(src_t)
+        assert np.allclose(
+            res_t.coord("altitude").points,
+            np.moveaxis(src_t.coord("altitude").points, 1, 2),
+        )
+        src_t.transpose([0, 2, 1])
+        res_t = rg_t(src_t)
+        assert np.allclose(
+            res_t.coord("altitude").points,
+            np.moveaxis(src_t.coord("altitude").points, 1, 2),
```

Comment:
The tests look good although I think they need to be restructured. You're calling the regridder three times here which seems to go against the ethos of unit tests. I would prefer a separate call to the regridder per unit test. Could you perhaps move the set up of src and tgt into a fixture?